checker.cpp

```diff
@@ -1,22 +1,72 @@
 #include <assert.h>
 #include <iostream>
+#include <string>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
+void toConsoleExtreamValue(string parameter_name)
+{
+  cout << parameter_name << "is extreamly low or high!\n";
+}
+
+void extreamValueCheck(float Parameter_value, string parameter_name, float lower_limit, float upper_limit)
+{
+ if((Parameter_value < lower_limit)||(Parameter_value > upper_limit))
+  {
+    toConsoleExtreamValue(parameter_name);
+  } 
+  else
+  {
+   //do nothing;
+  }
+}
+  
+void toConsole(float Parameter_value, string parameter_name)
+{
+  cout << parameter_name << "out of range!\n";
+  
+  if(parameter_name == "Temperature ")
+  {
+    extreamValueCheck(Parameter_value, parameter_name, -10, 100);
+  }
+  else if(parameter_name == "State of Charge ")
+  {
+    extreamValueCheck(Parameter_value, parameter_name, -10, 150);
+  } 
+  else
+  {
+       extreamValueCheck(Parameter_value, parameter_name, -5, 15);
+  }
+}
+
+
+bool isBatteryParameterInRange(float Parameter_value, float min, float max, string parameter_name)
+{
+  if((Parameter_value < min) || (Parameter_value > max))
+  {
+    toConsole(Parameter_value,parameter_name);
     return false;
+  } 
+  else
+  {
+    return true;
   }
-  return true;
+}
+
+
+bool batteryIsOk(float temperature, float soc, float chargeRate) {
+  
```

Comment:
Can we prevent complexity from increasing here, when more parameters are added? Think of how we state the problem: 'battery is not ok even if any one parameter is not ok'